python/ray/projects/scripts.py

```diff
@@ -338,5 +373,7 @@ def session_start(command, args, shell, name):
     "--name", help="Name of the session to run this command on", default=None)
 def session_execute(command, args, shell, name):
     runner = SessionRunner(session_name=name)
-    cmd = runner.format_command(command, args, shell)
-    runner.execute_command(cmd)
+    commands = runner.format_command(command, args, shell)
```

Comment:
This is a leaky abstraction. Because we are assuming `len(commands) > 1` means multiple sessions. 

One possible alternative:
1. parse the arguments and if wildcard detected, generate multiple calls to session start.
2. the semantic is now:
```
session start cmd_name --wheel {whl_url} --workload_name "*"
```
is equivalent to parallel calls
```
session start cmd_name --wheel {whl_url} --workload_name "pbt"
session start cmd_name --wheel {whl_url} --workload_name "apex"
session start cmd_name --wheel {whl_url} --workload_name "impala"
...
```

Reply:
I refactored the code to make it clearer. It does still use this pattern under the hood, but it should be ok now :)